src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component;
+use ILIAS\UI\Implementation\Component\Modal\LightboxImagePage;
+
+/**
+ * Interface Factory
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      An Interruptive Modal disrupts the user in critical situation,
+     *      forcing him or her to focus on the task at hand.
+	 *   composition: >
+	 *      The Modal states why this situation needs attention and may point out consequences.
+	 *   effect: >
+	 *      All controls of the original context are inaccessible until the Modal is completed.
+	 *      Upon completion the user returns to the original context.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *        Due to the heavily disruptive nature of this type of Modal it MUST be
+	 *        restricted to critical situations (e.g. loss of data).
+	 *     2: >
+	 *        All delete-actions are critical situations and SHOULD be implemented as an Interruptive Modal.
+	 *     3: >
+	 *        Interruptive Modals MUST contain a Default Button continuing the action that initiated the Modal
+	 *        (e.g. Delete the Item) on the left side of the footer of the modal and a button canceling
+	 *        the action on the right side of the footer as Default Button.
+	 *     4: >
+	 *        The Cancel Button in the footer and the Close Button in the header MUST NOT perform
+	 *        any additional action than closing the interruptive modal.
+	 * ---
+	 *
+	 * @param string $title
+	 * @param string $message
+	 * @param string $form_action
+	 *
+	 * @return \ILIAS\UI\Component\Modal\Interruptive
+	 */
+	public function interruptive($title, $message, $form_action);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     Interruptive items are displayed in an interruptive modal and represent the object(s) being affected
+	 *     by the critical action, e.g. deleting.
+	 *   composition:
+	 *   effect:
+	 * rules:
+	 * ---
+	 *
+	 * @param string $id
+	 * @param string $title
+	 * @param string $description
+	 *
+	 * @return InterruptiveItem
+	 */
+	public function interruptiveItem($id, $title, $description = '');
```

Comment:
My proposal would be to accept an optional icon in form of an Image object of the UI framework. We could then write an adapter for an ILIAS object to make it easier for the devs (and also offer a factory method), e.g:
```php
$course = new ilObjCourse(123);
$item = $factory->interruptiveILIASItem($course);
```